contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.2.0--2.3.0.sql

```diff
@@ -1284,1602 +671,5951 @@ LANGUAGE plpgsql
+
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_authid_login_ext', 'default_database_name');
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_authid_login_ext', 'default_language_name');
+
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_authid_user_ext', 'orig_username');
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_authid_user_ext', 'database_name');
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_authid_user_ext', 'default_schema_name');
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_authid_user_ext', 'default_language_name');
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_authid_user_ext', 'authentication_type_desc');
+
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_authid_user_ext_login_db_idx', 'database_name');
+-- we have to reindex babelfish_authid_user_ext_login_db_idx because given index includes database_name and we have to change its collation
+REINDEX INDEX sys.babelfish_authid_user_ext_login_db_idx;
+
+ALTER TABLE sys.babelfish_view_def ADD COLUMN create_date SYS.DATETIME, add COLUMN modify_date SYS.DATETIME;
+
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_view_def', 'schema_name');
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_view_def', 'object_name');
+
+-- primary key babelfish_view_def_pkey is based on sys.babelfish_view_def.schema_name and sys.babelfish_view_def.object_name
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_view_def_pkey', 'schema_name');
+CALL sys.babelfish_update_collation_to_default('sys', 'babelfish_view_def_pkey', 'object_name');
+
+CALL sys.babelfish_update_collation_to_default('sys', 'assemblies', 'name');
+CALL sys.babelfish_update_collation_to_default('sys', 'assemblies', 'clr_name');
+CALL sys.babelfish_update_collation_to_default('sys', 'assemblies', 'permission_set_desc');
+
+-- BABELFISH_FUNCTION_EXT
+CREATE TABLE sys.babelfish_function_ext (
+	nspname NAME NOT NULL,
+	funcname NAME NOT NULL,
+	orig_name sys.NVARCHAR(128), -- users' original input name
+	funcsignature TEXT NOT NULL COLLATE "C",
+	default_positions TEXT COLLATE "C",
+	flag_validity BIGINT,
+	flag_values BIGINT,
+	create_date SYS.DATETIME NOT NULL,
+	modify_date SYS.DATETIME NOT NULL,
+	PRIMARY KEY(nspname, funcsignature)
+);
+GRANT SELECT ON sys.babelfish_function_ext TO PUBLIC;
+
+SELECT pg_catalog.pg_extension_config_dump('sys.babelfish_function_ext', '');
+
+CREATE OR REPLACE FUNCTION sys.babelfish_get_pltsql_function_signature(IN funcoid OID)
+RETURNS text
+AS 'babelfishpg_tsql', 'get_pltsql_function_signature' LANGUAGE C;
+
+-- All the available views
+
+CALL sys.babelfish_update_collation_to_default('sys', 'sysdatabases', 'filename');
+
+CALL sys.babelfish_update_collation_to_default('sys', 'schemas', 'name');
+
+CALL sys.babelfish_update_collation_to_default('sys', 'server_principals', 'name');
```

Comment:
Don't we need to re-create dependent objects after updating collations?

Reply:
It is not necessary. Primary reason of re-creating dependent object would be because its column also pick the non-deterministic collations. But postgres, won't pick change in collation by merely re-creating the object. So we have to update catalog.